### PR TITLE
Aesthetical improvements + BB focus feature

### PR DESCRIPTION
**Pull request by SlimBenAmor**
In this PR, we have done the following:
 - Fix small bug and typo in `obviews.js`
 - Adapt Obviews styling and fonts to rocqstat-ng GUI
 - Add focus feature on a CFG block given its HTML ID or HEX address

PS: note that the font file should also be added in the environment/docker that run rocqstat-ng. A coming PR on [rocqstat-ng](https://github.com/statinf-software/rocqstat-ng) will fix this and integrate these updates.

Fixes: #16

### CODE REVIEW

**Review by jordr — COMMENTED** [review]
Thanks for the huge work with a lot of attention to detail.

image
image
image


The BB centering can make the experience... janky given that you're supposed to move around the CFG with drag and drop, to be debated with @IssaoSI: https://streamable.com/s2s30v

**Comment by Alex-Ralph** [comment]
This looks great so far, way better than it did before.

If I may offer a couple of suggestions: 
The font should be changed, the way that the current font displays some characters (such as `&` and `4`) is odd. We could either use archivo like the rest of the user interface, or a monospace font. If we have time I'd also like to try a monospace font for code snippets and archivo for menus etc.
image

The contents of some of the graph items aren't centered: 
image

And these buttons in the top should have their icons enlarged to take up more space, as it stands it's hard to tell what the icons are. 
image

Finally, the viewport should be anchored to the flow facts in some way, such that they can't disappear entirely off screen. It'd make it a bit harder to get yourself completely lost and thus need the "reset" button.

**Comment by SlimBenAmor** [comment]
Thank you @Alex-Ralph for the review:
I agree that `&` and `4` are weird but this is "Archivo" font. I tried to make it monospace for the code but it was not that easy because the code is parsed and colored character by character ([here](https://github.com/statinf-otawa/obviews/blob/9066f8e4e5e60d5cdfc7256b6dd1e10539e0ae5a/bin/obviews.py#L215-L233)) and add html-like tags for style these character is difficult because there are some character in the code snippet like `&nbsp`  that it is parsed character by character.

For the centering, the problem is that when graphviz generates the svg it uses the list of fonts defined in the OS/docker if "Archivo" does not exist it will use "times new roman" and the position (x,y) of each text tag are calculated according to characteristics (character size, spacing,...) of "times new roman" but in the rocqstat-ng gui it will use "Archivo" (with different character size, spacing) hence the problem of centering.

Good point for the button size I will increase them.

There is some problem with the focus feature that make the graph go very far or disappear but I removed it for the moment until I fix the problem in a new PR in order to avoid making this one blocked.

File: data/obviews/obviews.js [hunk 1]
```diff
@@ -177,17 +178,74 @@ function cfg_zoom(isZoomingIn=true, mouseEvt) {
 	cfg_transform();
 }
 
+// center on a given block by its address
```
Comment (jordr) [inline data/obviews/obviews.js:181]:
```suggestion
// center on a given block by its address (unused yet)
```

File: data/obviews/obviews.js [hunk 2]
```diff
@@ -177,17 +178,74 @@ function cfg_zoom(isZoomingIn=true, mouseEvt) {
...
+		CFG.cont.style.transition = `transform .5s`;
+		CFG.pos.x = 0;
+		CFG.pos.y = 0;
+		var old_scale = CFG.scale;
+		CFG.scale = CFG.default_scale;
+		var viewport_rect = document.getElementById("viewport").getBoundingClientRect();
+		var svg_rect = code.children[0].getBoundingClientRect();
+		var bb_rect = block_cont.getBoundingClientRect();
+		CFG.pos.x = (viewport_rect.width - bb_rect.width / old_scale) / 2 - (bb_rect.x - svg_rect.x) / old_scale;
+		if (bb_rect.height / old_scale > viewport_rect.height) {
```
Comment (jordr) [inline data/obviews/obviews.js:226]: Comment

**Review by jordr — CHANGES_REQUESTED** [review]
- View text needs to be greyed out when not available
- Some call nodes text overflow (PS: its fine if you make it square by my book)
- Entry/Exit nodes should be oval shape
- @IssaoSI smooth zoomin/zoomout tickles me but I am VERY subjective, can you check me on that, since @Alex-Ralph is on holidays?

**Comment by IssaoSI** [comment]
> * @IssaoSI smooth zoomin/zoomout tickles me but I am VERY subjective, can you check me on that, since @Alex-Ralph is on holidays?

Yeah, I talked about it with Slim and tried it myself, I don't feel triggered by it even if I can see where it is coming from: you're used to certain levels of zoom and not the in-between that's now available. I feel that the smoothness has an enhanced value for users who want another level of detail so if that's not a major block for you, I'll vote in favour of keeping it.

**Pull request merged**